### ISSUE TITLE
Windows error state audit

### ISSUE THREAD

**Issue by ermau** [xamarin/Xamarin.PropertyEditing#171]
## Design
- [ ] Windows
- [ ] Mac

We're not necessarily looking for a design on a per-editor basis, more of an general set of guidelines to apply whether that's a red outline, a popup, whatever.

## Apply to Elements:

- [ ] String
- [ ] Bool
- [ ] Property group
- Numerics
  - [ ] Int
  - [ ] Float
  - [ ] Byte
 - Brushes
    - [ ] Brush type selector
    - [ ] Solid color editor
 - Predefined values
    - [ ] Constrained
    - [ ] Flags
 - Resource Selector
   - [ ] Filters
   - [ ] Item navigation + selection
 - Type Selector
   - [ ] Filters
   - [ ] Item navigation + selection
 - Panel
   - [ ] Property filter
   - [ ] Arrange selector
   - [ ] Group navigation/expansion
   - [ ] Property button (ctrl+space)

**Comment by vancura** [comment]
What do you think about this?

![untitled-1](https://user-images.githubusercontent.com/4982/35975891-a1d081be-0cde-11e8-860d-e8f9716f3470.png)

**Comment by ermau** [comment]
I think that's the wrong platform, but even aside from that there's no column to the right of the property button and I don't want to introduce one for something that should basically never be seen.

**Comment by vancura** [comment]
> I think that's the wrong platform, but even aside from that there's no column to the right of the property button and I don't want to introduce one for something that should basically never be seen.

The column was there since ever just for this case, [see the original specs](https://mseng.visualstudio.com/_git/DevDivDesign?path=/spec/property-pad/README.md&version=GBmaster&createIfNew=true&anchor=metrics-and-fonts&_a=preview). It's the same place we have for the _Browse_ button (and other buttons).

**Comment by ermau** [comment]
We discussed not having that button as it moves resources interactions into two different places.

**Comment by vancura** [comment]
Ok, whatever :)

Then just put the warning icon into the textfield.

**Comment by BretJohnson** [comment]
Where did we end up here, on error reporting?
IMO, putting the warning icon in the textfield, like @vancura suggests, is good.
I'd lean toward left aligning it, though. That's what the current Android property editor does on VSMac, so left aligning is more consistent with current UX and makes the icon more visible to the user.

Also, I'd like to see this get in for 15.7, for Android.  On the Android designer VSMac side it really does provide a usability benefit in several scenarios, showing useful messages. We don't have to do it for 1st preview, but I'd rather like to see it for 15.7 GA, prioritized ahead of 15.8 only proppy features. Thanks.

**Comment by vancura** [comment]
There's one problem with left-aligning (in LTR text mode, obviously): the entered text would jump based on visibility of the warning icon. Imagine you have entered a valid value, then you add a character which invalidates it – the text jumps to make space for the icon. As soon as you fix the invalid character the icon hides, and text jumps left. This would be a bad UX in my opinion.

**Comment by BretJohnson** [comment]
@vancura Yeah, it would be bad if it behaved that way. But we only validate when the user leaves the control, not on every key press, so it isn't an issue.

Here's the current Android VSMac property UI showing the warning indicator, FYI:

![image](https://user-images.githubusercontent.com/245892/36544552-bdbbf83c-17b4-11e8-95fe-217950f66e03.png)

**Comment by BretJohnson** [comment]
And with the popup message itself showing:

![image](https://user-images.githubusercontent.com/245892/36544768-3f80b5ce-17b5-11e8-9be1-399711f61b76.png)

**Comment by vancura** [comment]
> But we only validate when the user leaves the control, not on every key press, so it isn't an issue.

Then it's fine of course.
